### PR TITLE
Add text for serializing numbers/lengths

### PR DESCRIPTION
**Pull request by nainar**
@shans  take a look? Will send it to @tabatkins after your ok

### CODE REVIEW

File: css-typed-om/Overview.bs [hunk 1]
```diff
@@ -2845,11 +2845,33 @@ of serializations of the contained {{CSSTransformComponent}} objects.
 {{CSSURLImageValue}} objects are serialized to the string given by
 "url(" + {{CSSURLImageValue/url}} + ")".
 
-{{CSSFontFaceValue}} Serialization {#fontfacevalue-serialization}
```
Comment (shans) [inline css-typed-om/Overview.bs]: Does this actually remove the FontFace serialization?

It's fine, we probably want to get rid of it, just strange to do both at once.

File: css-typed-om/Overview.bs [hunk 2]
```diff
@@ -2845,11 +2845,33 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
 
-{{CSSFontFaceValue}} objects are serialized to the value of their contained
-{{CSSFontFaceValue/fontFamilyName}}.
+This section covers how to serialize the various unit types in CSSOM
+
+: 'length'
+::
+    1. Let |values| initially be the empty [=list=].
+    2. If the <length> unit has a '-' sign character append it to |values|.
+    3. Append the result of serializing a 'number' to |values|.
```
Comment (shans) [inline css-typed-om/Overview.bs]: Eventually we'll want to make sure that there's links between all these things

File: css-typed-om/Overview.bs [hunk 3]
```diff
@@ -2845,11 +2845,33 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
-{{CSSFontFaceValue/fontFamilyName}}.
+This section covers how to serialize the various unit types in CSSOM
+
+: 'length'
+::
+    1. Let |values| initially be the empty [=list=].
+    2. If the <length> unit has a '-' sign character append it to |values|.
+    3. Append the result of serializing a 'number' to |values|. 
+    4. If the number is '0' skip this step. Else, append a unit identifier 
+        (a two-letter abbreviation) to |values|.
```
Comment (shans) [inline css-typed-om/Overview.bs]: which unit identifier?

File: css-typed-om/Overview.bs (hunk 1 above)
Comment (nainar) [inline css-typed-om/Overview.bs]: Done. This was a typo

File: css-typed-om/Overview.bs (hunk 2 above)
Comment (nainar) [inline css-typed-om/Overview.bs]: Done. Adding algorithm links

File: css-typed-om/Overview.bs (hunk 3 above)
Comment (nainar) [inline css-typed-om/Overview.bs]: Mentioned that this needs to be the length's unit identifier.

File: css-typed-om/Overview.bs [hunk 4]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+Units and Values Serialization {#unitsandvalues-serialization}
+--------------------------------------------------------------
+
+This section covers how to serialize the various unit types in CSSOM
+
+<div algorithm="serialize a 'length' value">
+
+To <dfn>serialize a 'length' value</dfn>, run these steps:
+    1. Let |values| initially be the empty [=list=].
+    2. If the <length> value has a '-' sign character append it to |values|.
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: Lengths don't have characters; look for it being negative instead, and then set it to its absolute value.

File: css-typed-om/Overview.bs [hunk 5]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
 {{CSSFontFaceValue/fontFamilyName}}.
 
+Units and Values Serialization {#unitsandvalues-serialization}
+--------------------------------------------------------------
+
+This section covers how to serialize the various unit types in CSSOM
+
+<div algorithm="serialize a 'length' value">
+
+To <dfn>serialize a 'length' value</dfn>, run these steps:
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: 'length' isn't a property. Same for all the others.

File: css-typed-om/Overview.bs [hunk 6]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+    1. Let |values| initially be the empty [=list=].
+    2. If the <length> value has a '-' sign character append it to |values|.
+    3. Append the result of [=serialize a <length> value=] to |values|.
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: What is this algorithm you're invoking? It looks like it's the same algorithm you're defining.  Maybe you meant to serialize the length's value as a number by invoking the "serialize a number" algorithm? (And if so, you want to keep the negative on there, as the number serializer will handle it for you.)

File: css-typed-om/Overview.bs [hunk 7]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+<div algorithm="serialize a 'length' value">
+
+To <dfn>serialize a 'length' value</dfn>, run these steps:
+    1. Let |values| initially be the empty [=list=].
+    2. If the <length> value has a '-' sign character append it to |values|.
+    3. Append the result of [=serialize a <length> value=] to |values|. 
+    4. If the number is '0' skip this step. Else, append the <length> 
+        value's unit identifier (a two-letter abbreviation) to |values|.
+    5. Return the result of concatenating all the items in |values|,
+        separated by " " (U+0020 SPACE).
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: You don't want space separation here. Just build a string directly by concatenation in each step; no need to indirect thru a list.

File: css-typed-om/Overview.bs [hunk 8]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+</div>
+
+<div algorithm="serialize a 'number' value">
+
+To <dfn>serialize a 'number' value</dfn>, run these steps:
+    1. Return the number respresented as a string.
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: If we're trying to replace CSSOM for this, we need to go into the same level of detail as it does, particularly regarding the minimum required amount of decimal precision, and if we ever output scinot.

File: css-typed-om/Overview.bs [hunk 9]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+</div>
+
+
+<div algorithm="serialize a 'percentage' value">
+
+To <dfn>serialize a 'percentage' value</dfn>, run these steps:
+    // TODO
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: Should just be "serialize a number" and then append a "%" to the end of the result, right?

File: css-typed-om/Overview.bs [hunk 10]
```diff
@@ -2847,10 +2847,49 @@ of serializations of the contained {{CSSTransformComponent}} objects.
...
+</div>
+
+
+<div algorithm="serialize a 'color' value">
+
+To <dfn>serialize a 'color' value</dfn>, run these steps:
+    // TODO
```
Comment (tabatkins) [inline css-typed-om/Overview.bs]: This one's gonna be quite complicated; you're not gonna want to have it in the same PR.

**Pull request closed without merging**